src/e2e-test/java/io.cdap.plugin/utils/ValidationHelper.java

```diff
@@ -0,0 +1,149 @@
+                                       List<Map<String, Object>> targetBigQueryRecords) {
+
+        String uniqueField = PluginPropertyUtils.pluginProp("primaryKey");
+        //Assert.assertEquals("Checking the size of lists ", targetBigQueryRecords.size(), sourceOracleRecords.size()); //doesn't make sense in our case, since few datatypes are dropped during assessment
+
+        // Logic to maintain the order of both lists and validate records based on that order.
+        Map BqUniqueIdMap = (Map)targetBigQueryRecords.stream()
+                .filter(t -> t.get("_is_deleted")==null)
+                .collect(Collectors.toMap(
+                        t -> t.get(uniqueField),
+                        t -> t,
+                        (x,y) -> {
+                            Long xSeqNum = Long.parseLong(x.get("_sequence_num").toString());
+                            Long ySeqNum = Long.parseLong(y.get("_sequence_num").toString());
+                            if(xSeqNum > ySeqNum){
+                                return x;
+                            }
+                            return y;
+                        }));
+
+
+        for (int record = 0; record < sourceOracleRecords.size(); record++) {
+            Map<String, Object> oracleRecord = sourceOracleRecords.get(record);
+            Object uniqueId = oracleRecord.get(uniqueField);
+            //EXP : In case data type not matched, convert to string and compare
+//            if( !uniqueId.getClass().equals(targetBigQueryRecords.get(0).get(uniqueField).getClass()) ){
+//                uniqueId = uniqueId.toString();
+//            }
+            Map<String, Object> bqRow = (Map<String, Object>) BqUniqueIdMap.get(uniqueId);
+            bqRow.remove("_is_deleted");
+            bqRow.remove("_sequence_num");
+
+            ValidationHelper.compareBothResponses( bqRow, oracleRecord);
+        }
+    }
+
+    public static void compareBothResponses(Map<String, Object> targetBigQueryRecords,
+                                            Map<String, Object> sourceOracleRecord) {
+        Assert.assertNotNull("Checking if target BigQuery Record is null", targetBigQueryRecords);
+        Set<String> bigQueryKeySet = targetBigQueryRecords.keySet();
+        for (String field : bigQueryKeySet) {
+            Object bigQueryFieldValue = targetBigQueryRecords.get(field);
+            Object sapFieldValue = sourceOracleRecord.get(field);
+//          if(!bigQueryFieldValue.getClass().equals(sapFieldValue.getClass())){
+//                    sapFieldValue = sapFieldValue.toString();
+//                    bigQueryFieldValue = bigQueryFieldValue.toString();
+//                }
+            Assert.assertEquals(String.format("Field %s is not equal: expected %s but got %s", field,
+                            sapFieldValue, bigQueryFieldValue),
+                    sapFieldValue, bigQueryFieldValue);
+        }
+    }
+
+
+
+
+    public static void waitForFlush() throws InterruptedException {
```

Comment:
I think we should move waitForFlush out of ValidationHelper to a more appropriate class

Reply:
yes. I was thinking the same but where should I move it. Should I create a new class?